dbt-adapters/src/dbt/include/global_project/macros/utils/equals.sql

```diff
@@ -0,0 +1,14 @@
+{% macro equals(expr1, expr2) %}
+    {{ return(adapter.dispatch('equals', 'dbt') (expr1, expr2)) }}
+{%- endmacro %}
+
+{% macro default__equals(expr1, expr2) -%}
+{%- if adapter.behavior.enable_truthy_nulls_equals_macro.no_warn %}
+    case when (({{ expr1 }} = {{ expr2 }}) or ({{ expr1 }} is null and {{ expr2 }} is null))
```

Comment:
next year we'll enable "extra truthy" nulls and support `or ({{ expr1 }} is '' and {{ expr2 }} is null)`